rero_ils/modules/items/serializers/collector.py

```diff
@@ -0,0 +1,378 @@
+        if issue.get('inherited_first_call_number') \
+                and not csv_data.get('item_call_number'):
+            csv_data['item_call_number'] = \
+                issue.get('inherited_first_call_number')
+        csv_data['issue_status'] = issue.get('status')
+        if issue.get('status_date'):
+            csv_data['issue_status_date'] = \
+                ciso8601.parse_datetime(
+                    issue.get('status_date')).date()
+        csv_data['issue_claims_count'] = \
+            issue.get('claims_count', 0)
+        csv_data['issue_expected_date'] = \
+            issue.get('expected_date')
+        csv_data['issue_regular'] = issue.get('regular')
+
+    @classmethod
+    def append_document_data(cls, csv_data, documents):
+        """Append document data.
+
+        :param csv_data: data dictionary.
+        :param documents: documents data.
+        """
+        try:
+            # update csv data with document
+            csv_data.update(documents.get(csv_data.get('document_pid')))
+        except Exception as err:
+            current_app.logger.error(
+                'ERROR in csv serializer: '
+                '{message} on document: {pid}'.format(
+                    message=err,
+                    pid=hit['document']['pid'])
+            )
+
+    @classmethod
+    def append_local_fields(cls, csv_data):
+        """Append local fields data.
+
+        :param csv_data: data dictionary.
+        """
+        def _append_res_local_fields(resource_type, resource_pid, csv_data):
+            """Append local fields data.
+
+            :param csv_data: data dictionary.
+            :param resource_type: resoruce_type.
+            :param resource_pid: resource_pid.
+            """
+            org_pid = csv_data['item_org_pid']
+            if local_fields := LocalField.get_local_fields_by_resource(
+                    resource_type, resource_pid, organisation_pid=org_pid):
+                for field, num in itertools.product(
+                        local_fields, range(1, 11)):
+                    field_name = f'{resource_type}_local_field_{num}'
+                    if field_data := field.get(
+                            'fields', {}).get(f'field_{num}'):
+                        csv_data[field_name] = \
+                            ' '.join(filter(None, field_data))
+
+        _append_res_local_fields('item', csv_data['item_pid'], csv_data)
+        _append_res_local_fields('doc', csv_data['document_pid'], csv_data)
+
+    @classmethod
+    def get_loans_by_item_pids(cls, item_pids=None):
+        """Get loans for the given item pid list.
+
+        :param item_pids: item pids.
+        :return dict of item_pids having the checkouts, renewals,
+                last_transaction_dates, last_checkouts data.
+        """
+        def add_pid(loans, pid):
+            """Add pid to loans."""
+            if pid not in loans:
+                loans[pid] = {}
+
+        # initial es query
```

Comment:
maybe a little bit explanation about this query could be nice. (with a sample output exemple ?)